libs/s25main/lua/LuaInterfaceGame.cpp

```diff
@@ -313,6 +319,37 @@ void LuaInterfaceGame::PostMessageWithLocation(int playerIdx, const std::string&
                                                       gw.MakeMapPoint(Position(x, y))));
 }
 
+namespace {
+void MarkCampaignChapter(const std::string& campaignUid, unsigned char chapter, char code)
+{
+    if(chapter < 1)
+        return;
+
+    auto& saveData = SETTINGS.campaigns.saveData[campaignUid];
+
+    if(saveData.length() < chapter)
+        saveData.resize(chapter);
+
+    saveData[chapter - 1] = code;
+}
+} // namespace
+
+void LuaInterfaceGame::EnableCampaignChapter(const std::string& campaignUid, unsigned char chapter)
+{
+    MarkCampaignChapter(campaignUid, chapter, CampaignSaveCodes::chapterEnabled);
```

Comment:
This should check if the campaign is not already completed. Otherwise a continent that is conquered can be marked as just enabled (not conquered) by conquering a different continent. Probably this check should be in `MarkCampaignChapter` (cannot set status to "less" than before).